streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -681,4 +701,18 @@ public int hashCode() {
         }
     }
 
+    public static class BackoffRecord {
+        private long attempts;
+        private long lastAttemptMs;
+        private static final ExponentialBackoff EXPONENTIAL_BACKOFF = new ExponentialBackoff(1, 2, 1000, 0.5);
+
+
+        public BackoffRecord() {
+            this.attempts = 0;
```

Comment:
shoudln't this initialize lastAttemptMs as well?

also, you seem to be creating this record on the first attempt. Then attempts should be set to 1, no?